zone/guild.cpp

```diff
@@ -317,36 +362,93 @@ void EntityList::SendGuildMembers(uint32 guild_id) {
 	while (it != client_list.end()) {
 		Client *client = it->second;
 		if (client->GuildID() == guild_id) {
-			client->SendGuildMembers();
+			//client->SendGuildMembers();
 		}
 		++it;
 	}
 }
 
-void EntityList::SendGuildList() {
-	auto it = client_list.begin();
-	while (it != client_list.end()) {
-		Client *client = it->second;
-		client->SendGuildList();
-		++it;
+void EntityList::SendGuildList()
+{
+	for (auto const& c : client_list) {
+		c.second->SendGuildList();
 	}
 }
 
-void Client::SendGuildJoin(GuildJoin_Struct* gj){
+void Client::SendGuildJoin(GuildJoin_Struct *gj)
+{
 	auto outapp = new EQApplicationPacket(OP_GuildManageAdd, sizeof(GuildJoin_Struct));
-	GuildJoin_Struct* outgj=(GuildJoin_Struct*)outapp->pBuffer;
-	outgj->class_ = gj->class_;
+	auto outgj  = (GuildJoin_Struct *) outapp->pBuffer;
+	outgj->class_   = gj->class_;
 	outgj->guild_id = gj->guild_id;
-	outgj->level = gj->level;
+	outgj->level    = gj->level;
+	outgj->rank     = gj->rank;
+	outgj->zoneid   = gj->zoneid;
 	strcpy(outgj->name, gj->name);
```

Comment:
I know this is unchanged but let's add a little protection since it's fixed size
`strn0cpy(outgj->name, gj->name, sizeof(outgj->name));`